packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -300,10 +305,12 @@ class CupertinoTextField extends StatefulWidget {
        assert(readOnly != null),
        assert(autofocus != null),
        assert(obscuringCharacter != null && obscuringCharacter.length == 1),
-       assert(obscureText != null),
+       assert(obscureText != null || obscureTextBehavior != null),
```

Comment:
We should also make sure that people don't try to use both:

```dart
assert(obscureText == null || obscureTextBehavior == null);
```

Here and everywhere else that accepts obscureTextBehavior.